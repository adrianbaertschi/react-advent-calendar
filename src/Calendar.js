import React from "react";
import {Day} from "./Day";
import './Calendar.css'

class Calendar extends React.Component {
    constructor(props, context) {
        super(props, context);
        this.dates = [];
        for (let i = 1; i < 25; i++) {
            const date = new Date(2019, 11, i);
            this.dates.push(date);
        }
    }

    render() {
        const listItems = this.dates.map((day) =>
            <Day key={day.getDate()} day={day}/>
        );
        return (
            <div className="calendar">
                <ul className="days">{listItems}</ul>
            </div>
        )
    }
}

export default Calendar;