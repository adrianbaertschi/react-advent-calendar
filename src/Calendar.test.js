import React from 'react';
import {render} from '@testing-library/react'
import Calendar from "./Calendar";
import renderer from 'react-test-renderer'

test('renders 24 fields', async () => {
    const {getAllByText} = render(<Calendar/>);
    expect(getAllByText(/\d/)).toHaveLength(24)
});

test('renders calendar like snapshot', async () => {
    const tree = renderer
        .create(<Calendar/>)
        .toJSON();
    expect(tree).toMatchSnapshot()
});