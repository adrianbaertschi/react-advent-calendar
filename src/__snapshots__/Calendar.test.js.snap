// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calendar like snapshot 1`] = `
<div
  className="calendar"
>
  <ul
    className="days"
  >
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="enabled"
      >
         
        1
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="enabled"
      >
         
        2
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="enabled"
      >
         
        3
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="enabled"
      >
         
        4
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        5
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        6
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        7
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        8
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        9
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        10
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        11
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        12
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        13
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        14
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        15
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        16
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        17
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        18
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        19
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        20
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        21
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        22
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        23
         
      </a>
    </li>
    <li
      className="day"
      onClick={[Function]}
    >
      <a
        className="disabled"
      >
         
        24
         
      </a>
    </li>
  </ul>
</div>
`;
